.github/workflows/ci.yml

```diff
@@ -0,0 +1,33 @@
+name: CI
+on:
+  push:
+    branches:
+      - "**"
+
+jobs:
+  lint-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Lint commit message
+        uses: wagoid/commitlint-github-action@v4
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      # Cache node_modules
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          cache: "npm"
+      - run: npm ci
+      - uses: nrwl/nx-set-shas@v3
+      # This line is needed for nx affected to work when CI is running on a PR
```

Comment:
@edgarkhanzadian - I've just realised this causes fails when PR's are merged to `dev`.

I'll try add a restriction to not run on `dev` / `main` but that's not really a good solution. Any ideas here? 

I'm thinking maybe we shouldn't run this job in CI and have individual jobs

![Screenshot 2023-11-29 at 07 01 25](https://github.com/leather-wallet/mono/assets/2938440/697fe67c-3537-4505-8d7e-38bdf6607df2)